npm/podController.go

```diff
@@ -252,34 +246,19 @@ func (c *podController) deletePod(obj interface{}) {
 		return
 	}
 
-	// (TODO): Reduce scope of lock later
-	c.npMgr.Lock()
-	defer c.npMgr.Unlock()
-
-	// If this pod object is not in the PodMap, we do not need to clean-up states for this pod
-	// since podController did not apply for any states for this pod
-	_, npmPodExists := c.npMgr.PodMap[key]
-	if !npmPodExists {
-		return
-	}
-
 	c.workqueue.Add(key)
 }
 
-func (c *podController) Run(threadiness int, stopCh <-chan struct{}) error {
+func (c *podController) Run(stopCh <-chan struct{}) {
 	defer utilruntime.HandleCrash()
 	defer c.workqueue.ShutDown()
 
-	klog.Infof("Starting Pod %d worker(s)", threadiness)
-	for i := 0; i < threadiness; i++ {
-		go wait.Until(c.runWorker, time.Second, stopCh)
-	}
+	klog.Infof("Starting Pod worker")
+	go wait.Until(c.runWorker, time.Second, stopCh)
```

Comment:
I think that was what I want sine it is a well-defined pattern in controller, but in the our meeting, this was suggested to prevent unexpected situation.

Reply:
Lets put a pin on this and discuss in next meeting.